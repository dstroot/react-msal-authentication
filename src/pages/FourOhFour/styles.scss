.cover {
  width: 100%;
  // padding: 60px 30px;

  /* Set a specific min height */
  min-height: 80vh;

  /* Position and center the image to scale nicely on all screens */
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  position: relative;

  .color-wash {
    /* Set a specific min height */
    min-height: 80vh;
    /* color overlay */
    background-color: rgba(0, 0, 0, 0.5);
  }
}
