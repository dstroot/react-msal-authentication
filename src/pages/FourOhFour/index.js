import React from "react";
import PropTypes from "prop-types";

import image from "./media/giphy.webp";
import "./styles.scss";

const FourOhFour = () => {
  return (
    <div className="cover" style={{ backgroundImage: `url(${image})` }}>
      <div className="d-flex justify-content-center color-wash">
        <h1 className="mt-5 p-5 display-3 text-white">Oops... 404</h1>
      </div>
    </div>
  );
};

// https://codeburst.io/validating-props-easily-with-react-proptypes-96e80208207

FourOhFour.propTypes = {
  image: PropTypes.string
};

FourOhFour.defaultProps = {
  image: "https://i.giphy.com/media/sMABNWsUWPC1y/giphy.gif"
};

export default FourOhFour;
