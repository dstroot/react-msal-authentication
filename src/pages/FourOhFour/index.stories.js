import React from 'react';
import { storiesOf } from '@storybook/react';

// components
import FourOhFour from '../FourOhFour';

// FourOhFour section
storiesOf('FourOhFour', module).add('show FourOhFour', () => <FourOhFour />);
