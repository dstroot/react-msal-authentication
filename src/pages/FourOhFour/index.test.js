import React from 'react';
import renderer from 'react-test-renderer';

// import component to test
import FourOhFour from '../FourOhFour';

describe('FourOhFour', () => {
  it('it should render', () => {
    const component = renderer.create(<FourOhFour />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
