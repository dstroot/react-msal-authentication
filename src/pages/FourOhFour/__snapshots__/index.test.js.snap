// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FourOhFour it should render 1`] = `
<div
  className="cover"
  style={
    Object {
      "backgroundImage": "url(giphy.webp)",
    }
  }
>
  <div
    className="d-flex justify-content-center color-wash"
  >
    <h1
      className="mt-5 p-5 display-3 text-white"
    >
      Oops... 404
    </h1>
  </div>
</div>
`;
